### PR TITLE
Fix docs

### PR DESCRIPTION
**Pull request by evanxg852000**
Fix some typos in the docs.
I might have added some debatable changes so I will explain as review comment

### CODE REVIEW

File: docs/overview/concepts/indexing.md [hunk 1]
```diff
@@ -7,11 +7,11 @@ sidebar_position: 1
 
 Quickwit ingests JSON records and refers to them as "documents" or "docs". Each document must be a JSON object. When ingesting files, documents must be separated by a newline.
 
-As of version 0.2, Quickwit does not support file formats such as `Avro` or `CSV`. Compression formats such as `bzip2` or `gzip` are also not supported.
```
Comment (evanxg852000) [inline docs/overview/concepts/indexing.md:10]: As a user what I understand with `As of version 0.2,`  is that it was previously supported and has been removed or changed.

File: docs/overview/concepts/indexing.md [hunk 2]
```diff
@@ -7,11 +7,11 @@ sidebar_position: 1
...
+Quickwit does not yet support file formats such as `Avro` or `CSV`. Compression formats such as `bzip2` or `gzip` are also not supported yet.
 
 ## Data model
 
-As of version 0.2, Quickwit only supports indexes with a fixed schema. The "document mapping" of an index, also commonly called "doc mapping", is a list of field names and types that declares the schema of an index. Additionally, a doc mapping specifies how documents are indexed (tokenizers) and stored (column- vs. row-oriented).
```
Comment (evanxg852000) [inline docs/overview/concepts/indexing.md:14]: ditto

File: docs/overview/concepts/indexing.md [hunk 3]
```diff
@@ -7,11 +7,11 @@ sidebar_position: 1
...
+So far, Quickwit only supports indexes with a fixed schema. The "document mapping" of an index, also commonly called "doc mapping", is a list of field names and types that declares the schema of an index. Additionally, a doc mapping specifies how documents are indexed (tokenizers) and stored (column-oriented vs. row-oriented).
```
Comment (fmassot) [inline docs/overview/concepts/indexing.md:14]:
```suggestion
Quickwit supports both schemaless indexes and fixed schemas. The "document mapping" of an index, also commonly called "doc mapping", is a list of field names and types that declares the schema of an index. For a schemaless or mixed fixed schema and schemaless indexing, follow our [guide on schemaless indexing](../../guides/schemaless.md). Additionally, a doc mapping specifies how documents are indexed (tokenizers) and stored (column-oriented vs. row-oriented).
```

**Review by fmassot — APPROVED** [review]
LGTM. Can you take into account my suggestion though? We support schemaless indexing so it's important to fix that.
Approved to ease the merge process.

**Pull request merged**